src/components/dialog/Dialog.tsx

```diff
@@ -93,9 +97,10 @@ const Dialog: GenericComponent<DialogProps> = ({
       <DialogBase.Footer
         display="flex"
         justifyContent={leftAction ? 'space-between' : 'flex-end'}
+        alignItems="center"
         className={cx({ [theme['scroll-shadow']]: !reachedScrollEnd && showScrollShadow })}
       >
-        {leftAction && <Button {...leftAction} />}
+        {leftAction && (leftAction.isToggle ? <Toggle {...leftAction} /> : <Button {...leftAction} />)}
```

Comment:
PP: I'd make a render prop out of these, because like @JorenSaeyTL already said, that would make it more future-proof.

```suggestion
        {renderLeftAction ? renderLeftAction(leftAction) : <Button {...leftAction} />)}
```

Reply:
Maybe we should first loop this back to design to see what is the purpose of the `leftAction` actually.
I'm not a big fan of just allowing you to render anything in this dialog which is supposed to force consistency.

Reply:
Edited my suggestion a bit, but I do agree with your point too.

Reply:
I agree, I don't think we should just allow anything.